### PR TITLE
Adding node-problem-detector stable chart

### PR DESCRIPTION
**Pull request by max-rocket-internet**
I would like to add a chart for the node-problem-detector:
https://github.com/kubernetes/node-problem-detector

This tool is already in the [cluster addons repo](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/node-problem-detector) for some time now.

### CODE REVIEW

**Comment by max-rocket-internet** [comment]
Any update @unguiculus ?

Or perhaps someone else? @scottrigby or @mgoodness or @davidkarlsen or @prydonius

**Comment by unguiculus** [comment]
Please compare with what you get from `helm create` using latest Helm and update accordingly. Make sure you follow our review guidelines and please apply our RBAC best practices.

https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md
https://github.com/helm/helm/blob/master/docs/chart_best_practices/rbac.md

**Comment by max-rocket-internet** [comment]
Hi @unguiculus!

`helm create` is not listed anywhere in the [Contributing Guidelines](https://github.com/helm/charts/blob/master/CONTRIBUTING.md). Nevertheless, I've run this command and compared it to what I have already. As such, I've added:
- Added `affinity` option
- Added `.helmignore` file
- `nameOverride` and `fullnameOverride` options
- Changed all labels

Would be cool if `helm create` produced the RBAC files.

It also seems to remove the `app` and `chart` labels and replaced it with `app.kubernetes.io/name` and `helm.sh/chart` which kind of breaks backwards compatibility? I've included both. Is that OK? This is confusing TBH.

> Make sure you follow our review guidelines

✅

> please apply our RBAC best practices.

OK I added a conditional around the service account and the `serviceAccountName` helper. I've compared it to 2 other popular charts, I think it's in line now.

File: stable/node-problem-detector/templates/clusterrolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    chart: {{ template "node-problem-detector.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    app.kubernetes.io/name: {{ include "node-problem-detector.name" . }}
+    helm.sh/chart: {{ include "node-problem-detector.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+subjects:
+- kind: ServiceAccount
+  name: {{ template "node-problem-detector.fullname" . }}
```
Comment (unguiculus) [inline stable/node-problem-detector/templates/clusterrolebinding.yaml]: `node-problem-detector.serviceAccountName`

File: stable/node-problem-detector/templates/daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+apiVersion: extensions/v1beta1
```
Comment (unguiculus) [inline stable/node-problem-detector/templates/daemonset.yaml]: Use `apps/v1`

File: stable/node-problem-detector/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+| `image.tag`                        | Image tag                       | `v0.5.0`                                                    |
+| `image.pullPolicy`                 | Image pull policy               | `IfNotPresent`                                              |
+| `rbac.create`                      | RBAC                            | `true`                                                      |
+| `resources.limits.cpu`             | CPU limit                       | `200m`                                                      |
+| `resources.limits.memory`          | Memory limit                    | `100Mi`                                                     |
+| `resources.requests.cpu`           | CPU request                     | `20m`                                                       |
+| `resources.requests.memory`        | Memory request                  | `20Mi`                                                      |
+| `tolerations`                      | Optional daemonset tolerations  | `{}`                                                        |
```
Comment (unguiculus) [inline stable/node-problem-detector/README.md]: The config table in incomplete.

File: stable/node-problem-detector/values.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  tag: v0.5.0
+  pullPolicy: IfNotPresent
+
+nameOverride: ""
+fullnameOverride: ""
+
+rbac:
+  create: true
+
+resources:
```
Comment (unguiculus) [inline stable/node-problem-detector/values.yaml]: Con' specify defaults for resources. Comment them out.

File: stable/node-problem-detector/values.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  limits:
+    cpu: "200m"
+    memory: "100Mi"
+  requests:
+    cpu: "20m"
+    memory: "20Mi"
+
+annotations: {}
+
+tolerations: {}
```
Comment (unguiculus) [inline stable/node-problem-detector/values.yaml]: `tolerations: []`

**Review by unguiculus — CHANGES_REQUESTED** [review]
There is a new convention for standard labels in Kubernetes land which Helm now follows. Please remove the old labels. Since this is a new chart, there are no backwards compatibility issues at all and we should follow the new convention.

**Comment by max-rocket-internet** [comment]
@unguiculus 

> There is a new convention for standard labels in Kubernetes land which Helm now follows. Please remove the old labels.

OK understood. I was copying older charts. Now all fixed.

**Comment by max-rocket-internet** [comment]
Any update @unguiculus ?

File: stable/node-problem-detector/README.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+## TL;DR;
+
+```console
+$ helm install stable/node-problem-detector
+```
+
+## Prerequisites
+
+- Kubernetes 1.8+ with Beta APIs enabled
```
Comment (unguiculus) [inline stable/node-problem-detector/README.md]: Should probably be 1.9+

File: stable/node-problem-detector/templates/_helpers.tpl [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "node-problem-detector.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/* Generate basic labels */}}
+{{- define "node-problem-detector.labels" }}
```
Comment (unguiculus) [inline stable/node-problem-detector/templates/_helpers.tpl]: Not used. Remove it.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by max-rocket-internet** [comment]
@unguiculus all done! Please re-review.

**Pull request merged**